### PR TITLE
Add 'skipMany' to skip values without needing to build an intermediat…

### PR DESCRIPTION
**Pull request by merijn**
I often find myself needing to skip over many tags while scraping messy data. Having a convenient combinator for this is nicer than repeated use of 'manyIgnoreYield (return Nothing)', although not absolutely crucial.

### CODE REVIEW

File: xml-conduit/Text/XML/Stream/Parse.hs [hunk 1]
```diff
@@ -978,6 +979,13 @@ many' :: MonadThrow m
       -> Consumer Event m [a]
 many' consumer = manyIgnore consumer ignoreAllTreesContent
 
+-- | Keep parsing elements as long as the parser returns 'Just', but throw the
+--   elements away without building an intermediate list.
+skipMany :: MonadThrow m
+         => Consumer Event m (Maybe a)
+         -> Consumer Event m ()
+skipMany consumer = manyIgnoreYield (return Nothing) ((() <$) <$> consumer)
```
Comment (snoyberg) [inline xml-conduit/Text/XML/Stream/Parse.hs]: Is `() <$` the same as `void`?

Comment (merijn) [inline xml-conduit/Text/XML/Stream/Parse.hs]: Yes, I think I initially tried `() <$` which doesn't map inside the Maybe and didn't think of using `void` when changing is, I've changed that now.

**Comment by k0ral** [comment]
How does `skipMany` differ (in behavior or in memory usage) from `void . many` ?

**Comment by merijn** [comment]
`void . many` builds up an intermediate list which is then discarded. Whereas this builds up no intermediate list, as a result this consumes O(1) space, vs O(n) for `void . many`, similar to the reason for `sequence_` existing in addition to `sequence`. This function would also be nice for cleaning up the issue with `takeAllTreesContent` mentioned in the other PR.

**Comment by merijn** [comment]
Actually, `many_` might be a better name (similar to `sequence` and `sequence_`) as this doesn't really "skip" anything per se (since the actions are still run)...

**Comment by k0ral** [comment]
Merged and renamed into `many_`. Thank you ! :)

**Pull request merged**